src/map/unit.cpp

```diff
@@ -2882,43 +2890,46 @@ int32 unit_unattackable(struct block_list *bl)
  * @param continuous: 
  *		0x1 - Whether or not the attack is ongoing
  *		0x2 - Whether function was called from unit_step_timer or not
- * @return Success(0); Fail(1);
+ * @return How the unit should stop; see e_unit_stop_walking
  */
 int32 unit_attack(struct block_list *src,int32 target_id,int32 continuous)
 {
 	struct block_list *target;
 	struct unit_data  *ud;
 	int32 range;
 
-	nullpo_ret(ud = unit_bl2ud(src));
+	if (ud = unit_bl2ud(src); ud == nullptr)
```

Comment:
Please dont put the assignment into the if, unless you declare the variable in this scope.